### PR TITLE
migrate to null safety

### PR DESCRIPTION
**Pull request by jakemac53**
This is almost a no-op migration, but in general we should still follow the guidelines to do a breaking change. Technically before you could have passed a `null` for `additionalGlobals` and it was handled, and that no longer is allowed.

### CODE REVIEW

**Comment by jakemac53** [comment]
@mbullington this can be immediately published - we are now in beta for null safety and have committed to no breaking changes in the feature.

You will get a warning on `pub lish` about the pre-release sdk dependency but can ignore it.

File: lib/preamble.dart [hunk 1]
```diff
@@ -124,7 +125,7 @@ if (dartNodeIsActuallyNode) {
 ///
 /// If [minified] is true, returns the minified version rather than the
 /// human-readable version.
-String getPreamble({bool minified: false, List<String> additionalGlobals: const []}) =>
+String getPreamble(
+        {bool minified: false, List<String> additionalGlobals: const []}) =>
     (minified ? _minified : _normal) +
-    (additionalGlobals == null ? "" :
-        additionalGlobals.map((global) => "self.$global=$global;").join());
+    (additionalGlobals.map((global) => "self.$global=$global;").join());
```
Comment (Mstrodl) [inline lib/preamble.dart:129]: I believe these changes need to be made to `tool/minify.js`. This is generated code.

**Review by Mstrodl — CHANGES_REQUESTED** [review]
Might be wrong, but I think things need to be modified in tool/minify.js

File: lib/preamble.dart (hunk 1 above)
Comment (jakemac53) [reply lib/preamble.dart:129]: oh, ok I am was just naively doing things here I can follow up (but will need to wait to tomorrow)

Comment (jakemac53) [reply lib/preamble.dart:129]: ~I see now that I probably need to update the actual generated javascript code as well?~ nevermind that looks hand written to me

Comment (Mstrodl) [reply lib/preamble.dart:129]: @jakemac53 maybe I don't understand, sorry. the `final _minified` here is generated by `tool/minified.js`. `lib/preamble.dart` is generated by `tool/minified.js`.

**Comment by mbullington** [comment]
Hi @jakemac53 `lib/preamble.js` is the "main" copy, then if you run `yarn && yarn minify` (or npm) we generate `preamble.dart` and `preamble.min.js`.

Yarn/Node.js scripts might seem odd for a Dart project, but it is so we can use Terser. 

Let me know if you need any more help. Thanks for contributing!

**Comment by jakemac53** [comment]
Hmm ok, I don't use node/yarn etc. I did push the change to `minify.js` though - would you be able to run the tool and push the generated output change here?

Otherwise I can try and figure out how to set up all of that.

File: lib/preamble.dart (hunk 1 above)
Comment (jakemac53) [reply lib/preamble.dart:129]: Ya no I figured it out :)

**Review by Mstrodl — APPROVED** [review]
I've run minify for you. I don't do Dart though so it's probably best to get someone who actually knows what they're talking about (wrt null safety) to merge this :P

**Comment by mbullington** [comment]
Happy to publish, however with @Mstrodl would like someone else with null-safety experience to check off first.

**Comment by jakemac53** [comment]
cc @natebosch for a review from a Dart null safety perspective?

Fwiw the change here is basically:

- Opt in to null safety by setting the min sdk to 2.12.0-0 (this changes your language version to 2.12 which is the first one that supports null safety). The `-0` part allows this package on any pre-release version (stable is not quite out yet). See the instructions here https://dart.dev/null-safety/migration-guide#migrating-by-hand which this follows.  
 - Since the `additionalGlobals` parameter is now marked as non-nullable (since it has a default it can be optional and non-nullable), the compiler would actually complain at you for even trying to null check it. So I have removed the explicit null handling which is no longer necessary.
 
Technically, this is a breaking change. Existing code could technically be passing an explicit `null` for that parameter, which would _override_ the default. When they upgrade to the latest version this will either be a compile time error now (if they are opted in to null safety) or it will be a runtime null dereference error (if they are opted out of null safety). It should be a rare case either way.

We could instead mark the parameter as nullable (add a `?` to the type), which would keep 100% compatibility but is probably not really the api you want longer term (it is pretty weird to have a nullable argument with a default).

**Review by natebosch — APPROVED** [review]

**Comment by jakemac53** [comment]
@mbullington @Mstrodl anything else you would like on my end? This is the last blocker for migrating the rest of `package:test` to null safety so it would be nice to get unblocked here :).

**Comment by Mstrodl** [comment]
LGTM

**Comment by Mstrodl** [comment]
You'll need @mbullington to publish though :)

**Comment by jakemac53** [comment]
Thanks :)

**Comment by jakemac53** [comment]
cc @mbullington any idea when you plan to publish?

**Comment by mbullington** [comment]
Can publish later today. Thanks for the contribution!

**Comment by mbullington** [comment]
@jakemac53 Published. Cheers on null safety 🎉

**Pull request merged**